File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        // Stack 객체 생성
        Stack<Integer> stack = new Stack<Integer>();
        
		for(int i : arr) {
			if(stack.isEmpty()) {
				stack.push(i);
			}else if(stack.peek() != i) {
				stack.push(i);
			}
		}
        
		int[] answer = new int[stack.size()];
        
		for(int i = 0; i < stack.size(); i++) {
			answer[i] = stack.get(i);
		}
        
		return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
const solution = (arr) => {
    const answer = [];
    
    arr.forEach((item, i) => {
        if(i === 0) {
            answer.push(item);
        } else {
            if(answer[answer.length - 1] !== item) {
                answer.push(item);
            }
        }
    })
    
    return answer;
}